Include/cpython/pystate.h

```diff
@@ -222,7 +222,7 @@ struct _ts {
 #  ifdef __wasi__
 #    define C_RECURSION_LIMIT 500
 #  else
-#    define C_RECURSION_LIMIT 800
+#    define C_RECURSION_LIMIT 2400
```

Comment:
This could do with a comment about the duplicating in test.support (possibly in the longer comment before this block, which GH won't let me edit in a suggestion.)